operator/pkg/remote/kyma_synchronization_context.go

```diff
@@ -166,48 +200,57 @@ func (c *KymaSynchronizationContext) SynchronizeRemoteKyma(ctx context.Context,
 		remoteKyma.Status.ObservedGeneration = remoteKyma.GetGeneration()
 		if err := c.runtimeClient.Status().Update(ctx, remoteKyma); err != nil {
 			recorder.Event(c.controlPlaneKyma, "Warning", err.Error(), "could not update runtime kyma status")
+
 			return true, err
 		}
 
 		c.controlPlaneKyma.Spec = remoteKyma.Spec
 		if err := c.controlPlaneClient.Update(ctx, c.controlPlaneKyma); err != nil {
 			recorder.Event(c.controlPlaneKyma, "Warning", err.Error(), "could not update control clane kyma")
+
 			return true, err
 		}
 
 		// as we have updated the control plane, we will requeue the object
 		return true, nil
-	} else if c.controlPlaneKyma.Status.ObservedGeneration != c.controlPlaneKyma.GetGeneration() {
+	}
+
+	if c.controlPlaneKyma.Status.ObservedGeneration != c.controlPlaneKyma.GetGeneration() {
 		// control plane got updated, runtime on cluster is using the wrong base instance for customization
+		//nolint:godox
 		// TODO this now requires custom merge logic, but for now we reapply the control plane version
 		remoteKyma.Spec = c.controlPlaneKyma.Spec
 	} else if remoteKyma.Status.State != c.controlPlaneKyma.Status.State {
```

Comment:
Hi @jakobmoellersap, can you double check if here is an `else if` or another `if` check? By reading the code, it seems should be another `if` check, because the condition is not same as before, but I'm not sure your initial thoughts.

Reply:
please do not make this to an if, otherwise the logic will try to first update the status and then the actual object. its meant to use 2 reconcile loops here. So thats why theres an elif control flow. I think we can rewrite it but then the code needs to be structured differently